implementations/rust/ockam/ockam_api/src/nodes/service/portals.rs

```diff
@@ -55,7 +55,7 @@ impl NodeManager {
 
         // Check that there is no entry in the registry with the same alias
         if self.registry.outlets.contains_key(&alias) {
```

Comment:
As you can see [here](https://github.com/build-trust/ockam/blob/1918c2849ecdc656ebf1d5796c073b22ced294d0/implementations/rust/ockam/ockam_app/src/shared_service/tcp_outlet/create.rs#L50) the alias is `None` so, we are currently allowing the user to create different outlets with the same `worker_addr` and/or same `socket_addr`. Those are the two cases we want to handle explicitly, similarly to what we currently do checking for duplicated `alias`.

As for the error messages, instead of using the `"A TCP outlet with name {name} already exists", perhaps you could write something like:

```rust
warn!("A TCP outlet with name {name} already exists");
let message = format!("name {name} already exists");
return Err(ockam_core::Error::new(...));
```

so that, in the `ockam_app` you can add the prefix `Failed to create service: {e}`.